worker/src/main/scala/org/apache/celeborn/service/deploy/worker/FetchHandler.scala

```diff
@@ -341,16 +343,19 @@ class FetchHandler(
   def handleEndStreamFromClient(streamId: Long, streamType: StreamType): Unit = {
     streamType match {
       case StreamType.ChunkStream =>
+        incChunkAppConnectionCounter(streamId)
         val (shuffleKey, fileName) = chunkStreamManager.getShuffleKeyAndFileName(streamId)
         getRawFileInfo(shuffleKey, fileName).closeStream(streamId)
       case StreamType.CreditStream =>
+        incCreditAppConnectionCounter(streamId)
         creditStreamManager.notifyStreamEndByClient(streamId)
       case _ =>
         logError(s"Received a PbBufferStreamEnd message with unknown type $streamType")
     }
   }
 
   def handleReadAddCredit(credit: Int, streamId: Long): Unit = {
+    incCreditAppConnectionCounter(streamId)
```

Comment:
This only happened when a connection is established. So here should not increase this counter.